src/absil/illib.fs

```diff
@@ -280,7 +280,6 @@ module List =
         | h::t -> if f h then Some (h,n) else findi (n+1) f t
 
     let chop n l = 
-        if n = List.length l then (l,[]) else // avoids allocation unless necessary 
```

Comment:
List.length is deadly, I wonder if some of the places shouldn't use array, in particular the things that we tend to look at list length from.